statement.go

```diff
@@ -146,17 +147,21 @@ func (s *SelectStatement) Execute(session *Session) (*Result, error) {
 	stmt := spanner.NewStatement(s.Query)
 	var iter *spanner.RowIterator
 
+	result := &Result{}
+
 	if session.InRwTxn() {
 		iter = session.rwTxn.QueryWithStats(session.ctx, stmt)
 	} else if session.InRoTxn() {
 		iter = session.roTxn.QueryWithStats(session.ctx, stmt)
+		result.Timestamp, _ = session.roTxn.Timestamp()
 	} else {
-		iter = session.client.Single().QueryWithStats(session.ctx, stmt)
```

Comment:
I tested
```
txn := session.client.Single()
defer txn.Close()
iter = txn.QueryWithStats(session.ctx, stmt)
if ts, err := txn.Timestamp(); err != nil {
	return nil, err
} else {
	result.Timestamp = ts
}
```
and got
```
ERROR: spanner: code = "Internal", desc = "read timestamp is unavailable"
```

I don't yet know a single transaction can't return the read timestamp but it become zero value of `time.Time` in real.
https://github.com/googleapis/google-cloud-go/blob/spanner/v1.4.0/spanner/transaction.go#L673-L675
It can be a bug of `google-cloud-go`.

Reply:
Add comment
https://github.com/cloudspannerecosystem/spanner-cli/pull/30/commits/15892890082bede530870b12dd5ee6ec6691dde6

Reply:
I have confirmed that singleUse read only transaction should be able to get the read timestamp.
```
$ jq -n '{transaction: {singleUse: {readOnly: {returnReadTimestamp: true}}}, sql: "SELECT 1"}'| curl -H "Authorization: Bearer $(gcloud auth print-access-token)" -s "https://spanner.googleapis.com/v1/${SESSION}:executeSql" -H "content-type: application/json" --data "@-" 
{
  "metadata": {
    "rowType": {
      "fields": [
        {
          "type": {
            "code": "INT64"
          }
        }
      ]
    },
    "transaction": {
      "readTimestamp": "2020-03-30T15:57:23.823503Z"
    }
  },
  "rows": [
    [
      "1"
    ]
  ]
}
```
So I am doubting `google-cloud-go`.

Reply:
I have tried debugger and set breakpoint to all write of  and I have noticed it is written only by `begin()` which isn't called in `aquireSingleUse()`.
https://github.com/googleapis/google-cloud-go/blob/spanner/v1.2.0/spanner/transaction.go#L459
https://github.com/googleapis/google-cloud-go/blob/spanner/v1.2.0/spanner/transaction.go#L478-L571

Reply:
I understood it is documented behavior of `ReadOnlyTransaction.Timestamp()`.
https://pkg.go.dev/cloud.google.com/go/spanner?tab=doc#ReadOnlyTransaction.Timestamp
> The value can only be read after some read or query has either returned some data or completed without returning any data.
I am wondering why it is filled in `ReadOnlyTransaction()` but not `Single()`

Reply:
Fixed in https://github.com/cloudspannerecosystem/spanner-cli/pull/30/commits/4bf24e24cbc7d41acb9e41fd2f4d01d8584a3757.

Reply:
I understood it is because `beginTransaction` is hidden into `acquireSingle` which is called in `prepareExecuteSQL`.
Executing `BEGIN RO` don't call `beginTransaction`.
It is unintuitive!

Reply:
Thank you for debugging in detail.

Yeah it looks like the read timestamp for single-read is set on this line.
https://github.com/googleapis/google-cloud-go/blob/master/spanner/read.go#L141

Reply:
Related discussion: https://twitter.com/furuyamayuki/status/1244810319588810752